taxonomy/models.py

```diff
@@ -0,0 +1,51 @@
+"""Models for lore taxonomy application"""
+from __future__ import unicode_literals
+
+from django.db import models
+
+from learningresources.models import (
+    Repository,
+    LearningResourceType,
+    LearningResource,
+)
+
+
+class Vocabulary(models.Model):
+    """Model for vocabulary table"""
+    MANAGED = "m"
+    FREE_TAGGING = "f"
+
+    repository = models.ForeignKey(Repository, on_delete=models.PROTECT)
+    name = models.TextField()
+    description = models.TextField()
+    required = models.BooleanField()
+    vocabulary_type = models.CharField(max_length=1, choices=(
+        (MANAGED, "managed"),
+        (FREE_TAGGING, "free tagging")
+    ))
+    weight = models.IntegerField()
+
+    learning_resource_types = models.ManyToManyField(
+        LearningResourceType,
+        related_name="vocabularies"
+    )
+
+    # pylint: disable=missing-docstring,no-init,too-few-public-methods,
+    # pylint: disable=old-style-class
+    class Meta:
+        unique_together = (("repository", "name"),)
+
+
+class Term(models.Model):
+    """Model for term table"""
+    vocabulary = models.ForeignKey(Vocabulary, on_delete=models.PROTECT)
+    label = models.TextField()
+    weight = models.IntegerField()
+
+    learning_resources = models.ManyToManyField(LearningResource,
+                                                related_name="terms")
+
+    # pylint: disable=missing-docstring,no-init,too-few-public-methods,
+    # pylint: disable=old-style-class
```

Comment:
I'm torn, but ok with either globally disabling, too-few-public-methods or no-init, or at least scoping those to the top of the models file.  I generally like the `too-few-public-methods` when making new classes, so I would hate to see it go, but we are mostly doing inheritance in this project anyway, so I don't think it would hurt us very bad to disable it all together. The missing-docstring however, should actually be scoped to the meta class though, as opposed to the root model class, but not a big deal, and definitely shouldn't be added to the top or global disable list.